generics/middleware/authenticator.js

```diff
@@ -417,8 +509,67 @@ module.exports = async function (req, res, next) {
       }
     }
   }
+
+
+    if (performInternalAccessTokenCheck) {
+      // Validate the presence of required headers
+      let adminOrgId = req.get('orgId');
+      let adminTenantId = req.get('tenantId');
+      let roles = decodedToken.data.roles;
+      let isAdmin = req.get('admin-auth-token') === process.env.ADMIN_AUTH_TOKEN;
+      let isTenantAdmin = checkForRole(roles, 'tenant_admin');
+      let isOrgAdmin = checkForRole(roles, 'org_admin');
+
+      if (isAdmin) {
+        if (!adminOrgId || !adminTenantId) {
```

Comment:
@borkarsaish65 what happen if super admin try to access the consumption api? it will fail right, you are expecting every api call he makes should have orgId and tenantId

Reply:
I have checked the flow and with current logic it will work. 
As I am doing internalapis check as well here so it wont fail consumption side apis